sample-operators/mysql-schema/src/main/java/io/javaoperatorsdk/operator/sample/MySQLSchemaReconciler.java

```diff
@@ -26,42 +27,23 @@
         @Dependent(type = SchemaDependentResource.class)
     })
 public class MySQLSchemaReconciler
-    implements Reconciler<MySQLSchema>, ErrorStatusHandler<MySQLSchema>,
-    ContextInitializer<MySQLSchema> {
+    implements Reconciler<MySQLSchema>, ErrorStatusHandler<MySQLSchema> {
 
-  static final String SECRET_FORMAT = "%s-secret";
-  static final String USERNAME_FORMAT = "%s-user";
-
-  static final String MYSQL_SECRET_NAME = "mysql.secret.name";
-  static final String MYSQL_SECRET_USERNAME = "mysql.secret.user.name";
-  static final String MYSQL_SECRET_PASSWORD = "mysql.secret.user.password";
-  static final String BUILT_SCHEMA = "built schema";
   static final Logger log = LoggerFactory.getLogger(MySQLSchemaReconciler.class);
 
   public MySQLSchemaReconciler() {}
 
-  @Override
-  public void initContext(MySQLSchema primary, Context context) {
-    final var name = primary.getMetadata().getName();
-    final var password = RandomStringUtils
-        .randomAlphanumeric(16); // NOSONAR: we don't need cryptographically-strong randomness here
-
-    final var secretName = String.format(SECRET_FORMAT, name);
-    final var userName = String.format(USERNAME_FORMAT, name);
-
-    // put information in context for other dependents and reconciler to use
-    context.put(MYSQL_SECRET_PASSWORD, password);
-    context.put(MYSQL_SECRET_NAME, secretName);
-    context.put(MYSQL_SECRET_USERNAME, userName);
-  }
 
   @Override
   public UpdateControl<MySQLSchema> reconcile(MySQLSchema schema, Context context) {
     // we only need to update the status if we just built the schema, i.e. when it's present in the
     // context
-    return context.get(BUILT_SCHEMA, Schema.class).map(s -> {
-      updateStatusPojo(schema, context.getMandatory(MYSQL_SECRET_NAME, String.class),
-          context.getMandatory(MYSQL_SECRET_USERNAME, String.class));
+    Secret secret = context.getSecondaryResource(Secret.class).orElseThrow();
+    SchemaDependentResource schemaDependentResource = context.managedDependentResourceContext()
+        .getDependentResource(SchemaDependentResource.class);
```

Comment:
I think it'd be better to get the dependent via its class in a way similar to what we do for secondary resources, i.e. `getDependentResource(Schema.class)`, though I guess this way this side-steps the issue of what to do when there are multiple dependents with the same type? Maybe provide both versions?

Reply:
> I think it'd be better to get the dependent via its class in a way similar to what we do for secondary resources, i.e. getDependentResource(Schema.class)

That is how it's done, just hidden behind dedicated context. Or?

>  though I guess this way this side-steps the issue of what to do when there are multiple dependents with the same type? Maybe provide both versions?

I intentionally did not cover this, since on denepnds_on we plan to have names of the resources, so I would add that as part of it?